optimum/neuron/modeling_seq2seq.py

```diff
@@ -0,0 +1,912 @@
+                input_ids.to("cpu")[:, :cur_len],
+                next_token_scores.to("cpu"),
+                next_tokens.to("cpu"),
+                next_indices.to("cpu"),
+                pad_token_id=pad_token_id,
+                eos_token_id=eos_token_id,
+                beam_indices=beam_indices,
+            )
+
+            beam_scores = beam_outputs["next_beam_scores"]
+            beam_next_tokens = beam_outputs["next_beam_tokens"]
+            beam_idx = beam_outputs["next_beam_indices"]
+
+            update_indices = torch.stack(
+                [torch.arange(batch_beam_size), torch.tensor(cur_len - 1).repeat(batch_beam_size)], dim=-1
+            )
+            update_indices_2 = torch.stack(
+                [torch.arange(batch_beam_size), torch.tensor(cur_len).repeat(batch_beam_size)], dim=-1
+            )
+            # First select beam_indices
+            device = input_ids.device
+            beam_idx_device = beam_idx.to(device=input_ids.device)
+            input_ids[:, :] = input_ids[beam_idx_device.long(), :]
+
+            # Then append new tokens
+            input_ids[update_indices_2[:, 0], update_indices_2[:, 1], None] = (
+                beam_next_tokens.unsqueeze(-1).to(device).to(torch.long)
+            )
+            input_ids = input_ids * 1  # Hack to materialize tensor
+
+            # update generated ids, model inputs, and length for next step
+            model_kwargs = self._update_model_kwargs_for_xla_generation(
+                model_kwargs,
+                batch_size=batch_beam_size,
+                is_encoder_decoder=self.config.is_encoder_decoder,
+                max_length=stopping_criteria.max_length,
+                seq_length=cur_len,
+                use_cache=model_kwargs["use_cache"],
+            )
+            if model_kwargs["past_key_values"] is not None:
+                model_kwargs["past_key_values"] = self._reorder_cache(
+                    model_kwargs["past_key_values"], beam_idx.to(torch.int64)
+                )
+
+            if return_dict_in_generate and output_scores:
+                beam_indices = tuple((beam_indices[beam_idx[i]] + (beam_idx[i],) for i in range(len(beam_indices))))
+
+            # increase cur_len
+            cur_len = cur_len + 1
+
+            # stop when each sentence is finished, or if we exceed the maximum length
+            stop_criterion_1 = beam_scorer.is_done
+            if isinstance(stopping_criteria, list):
+                if len(stopping_criteria) == 1:
+                    stopping_criteria = stopping_criteria[0]
+
+            # Cases that can be handled in XLA without requiring
+            # non-padded input_ids
+            if isinstance(stopping_criteria, MaxLengthCriteria):
+                stop_criterion_2 = cur_len >= stopping_criteria.max_length
+            elif isinstance(stopping_criteria, MaxTimeCriteria):
+                stop_criterion_2 = stopping_criteria(input_ids, scores)
```

Comment:
scores is never set, so it should be either `()` or `None`: is that expected ?

Reply:
yeah, because we are not calculating `next_token_scores_processed` with `logits_processor`, the `scores` is a little bit dummy here. But I would leave it this way, as we might support `logits_processor` during the compilation in the future.